### PR TITLE
test: add tests to verify setting cookie domain option works correctly

### PR DESCRIPTION
**Pull request by kevinpagtakhan**
### Summary

Add a test to verify that
1. default cookie domain is set correctly
2. custom cookie domain is set correctly

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

File: packages/analytics-browser/test/config.test.ts [hunk 1]
```diff
@@ -159,6 +161,7 @@ describe('config', () => {
           },
           sessionTimeout: 1,
           cookieOptions: {
+            domain: '.amplitude.com',
```
Comment (yuhao900914) [inline packages/analytics-browser/test/config.test.ts:164]: Should we give the domain and getTopLevelDomain a different value?

Comment (kevinpagtakhan) [reply packages/analytics-browser/test/config.test.ts:164]: what do you mean? what gap does it address?

Comment (yuhao900914) [reply packages/analytics-browser/test/config.test.ts:164]: if getTopLevelDomain and cookieOption have the same value, we are not sure the expected value comes from where.

Comment (kevinpagtakhan) [reply packages/analytics-browser/test/config.test.ts:164]: the mock result for `getTopLevelDomain` is only valid once and only for the earlier test block. the mock is no longer in effect on this test block. without a valid mock the `getTopLevelDomain` returns an empty string on a test environment.

**Review by yuhao900914 — APPROVED** [review]

File: packages/analytics-browser/test/config.test.ts (hunk 1 above)
Comment (kevinpagtakhan) [reply packages/analytics-browser/test/config.test.ts:164]: in addition, mocks are also restored after each test block to ensure the mock behavior does not leak onto the next test block as configured here: https://github.com/amplitude/Amplitude-TypeScript/blob/c5c3d62cf505e3df949a4225e3fa3ae2b56d5a0a/jest.config.js#L10

**Pull request merged**